File: SWEA/sol/6907.py
# 6907. 단위 변환기 프로그램 D4
# https://swexpertacademy.com/main/code/problem/problemDetail.do?contestProbId=AWh4GaY6EkEDFAXp&categoryId=AWh4GaY6EkEDFAXp&categoryType=CODE
# python 제출이 없음....
# example cass 는 pass했으나 test case는 python 미지원으로 불가
# C code로 porting 해 pass ( 6907.c )

si={"yotta" : 24,"zetta" : 21,"exa" : 18,"peta" : 15,"tera" : 12,"giga" : 9,"mega" : 6,"kilo" : 3,"hecto" : 2,"deca" : 1,"deci" : -1,\
"centi" : -2,"milli" : -3,"micro" : -6,"nano" : -9,"pico" : -12,"femto" : -15,"ato" : -18,"zepto" : -21,"yocto" : -24,"none": 0 }
for tc in range(1,int(input())+1):
    i=input().split()
    n=i[0]
    if len(i)==3:
        s1,s2=i[1],i[2]
    else:
        s1="none"
        s2=i[1]
    if float(n)>=1 and float(n)<10:
        temp=0
        pass
    elif n[0]=='0':
        dot=0
        for i in range(len(n)):
            if n[i]!="0" and n[i]!=".":
                break
        temp=1-i
        n=n[i:]
    else :
        dot=len(n)
        for i in range(len(n)):
            if n[i]==".":
                dot=i
                break
        n=n.replace(".","")
        n=n[0]+"."+n[1:]
        temp=dot-1
    print(f"#{tc} {n} * 10^{si[s1]+temp} {s2}")

File: SWEA/sol/6907.c
// 6907. 단위 변환기 프로그램 D4
// https://swexpertacademy.com/main/code/problem/problemDetail.do?contestProbId=AWh4GaY6EkEDFAXp&categoryId=AWh4GaY6EkEDFAXp&categoryType=CODE

#include <stdio.h>
#include <string.h>
int main(void)
{
	int test_case;
	int T,i,j,dot,temp,si,check;
    char s1[10], s2[51], n[1001];
	setbuf(stdout, NULL);
	scanf("%d", &T);

	for (test_case = 1; test_case <= T; ++test_case)
	{
        scanf(" %s %s",n,s1);
        check=1;
        if (strcmp(s1,"yotta")==0)si=24;
        else if (strcmp(s1,"zetta")==0)si=21;
        else if (strcmp(s1,"exa")==0)si=18;
        else if (strcmp(s1,"peta")==0)si=15;
        else if (strcmp(s1,"tera")==0)si=12;
        else if (strcmp(s1,"giga")==0)si=9;
        else if (strcmp(s1,"mega")==0)si=6;
        else if (strcmp(s1,"kilo")==0)si=3;
        else if (strcmp(s1,"hecto")==0)si=2;
        else if (strcmp(s1,"deca")==0)si=1;
        else if (strcmp(s1,"deci")==0)si=-1;
        else if (strcmp(s1,"centi")==0)si=-2;
        else if (strcmp(s1,"milli")==0)si=-3;
        else if (strcmp(s1,"micro")==0)si=-6;
        else if (strcmp(s1,"nano")==0)si=-9;
        else if (strcmp(s1,"pico")==0)si=-12;
        else if (strcmp(s1,"femto")==0)si=-15;
        else if (strcmp(s1,"ato")==0)si=-18;
        else if (strcmp(s1,"zepto")==0)si=-21;
        else if (strcmp(s1,"yocto")==0)si=-24;
        else{
            si=0;
            check=0;
        }
        if (check==1){
            scanf(" %s",s2);
        }
        else{
            strcpy(s2,s1);
            s1[0]='\0';
        }
        if(n[0]==48){
            dot=0;
            for(i=0;i<strlen(n);++i){
                if (n[i]!=48 && n[i]!=46) break;
            }
            temp=1-i;
            n[0]=n[i];
            if (strlen(n)>i+1){
                n[1]=46;
                for(j=i+1;j<=strlen(n);++j){
                    n[j-i+1]=n[j];
                }
            }
            else n[1]='\0';
        }
        else{
            dot=strlen(n);
            check=0;
            for (i=0;i<strlen(n);++i){
                if (n[i]==46){
                    dot=i;
                    check=1;
                    break;
                }
            }
            temp=dot-1;
            for (i=dot;i>1;--i){
                n[i]=n[i-1];
            }
            n[1]=46;
            if (check==0){
                n[dot+1]='\0';
            }
        }
        printf("#%d %s * 10^%d %s\n",test_case,n,si+temp,s2);
	}
    return 0;
}